tests/src/test/scala/com/nvidia/spark/rapids/unit/DecimalUnitTest.scala

```diff
@@ -0,0 +1,180 @@
+      }
+      val long = dec.toLong
+      withResource(GpuScalar.from(long, DecimalType(dec.precision, 0))) { s =>
+        s.getType.getScale shouldEqual 0
+        GpuScalar.extract(s).asInstanceOf[Decimal].toLong shouldEqual long
+      }
+    }
+    // test exception throwing
+    assertThrows[IllegalStateException] {
+      withResource(GpuScalar.from(true, DecimalType(10, 1))) { _ => }
+    }
+    assertThrows[IllegalArgumentException] {
+      val bigDec = Decimal(BigDecimal(Long.MaxValue / 100, 0))
+      withResource(GpuScalar.from(bigDec, DecimalType(15, 1))) { _ => }
+    }
+  }
+
+  test("test decimal as column vector") {
+    withResource(
+      GpuColumnVector.from(ColumnVector.fromDecimals(dec32Data.map(_.toJavaBigDecimal): _*),
+        DecimalType(DType.DECIMAL32_MAX_PRECISION, 5))) { cv: GpuColumnVector =>
+
+      cv.getRowCount shouldEqual dec32Data.length
+      val (precision, scale) = cv.dataType() match {
+        case dt: DecimalType => (dt.precision, dt.scale)
+      }
+      withResource(cv.copyToHost()) { hostCv: RapidsHostColumnVector =>
+        dec32Data.zipWithIndex.foreach { case (dec, i) =>
+          val rescaled = dec.toJavaBigDecimal.setScale(scale)
+          hostCv.getInt(i) shouldEqual rescaled.unscaledValue().intValueExact()
+          hostCv.getDecimal(i, precision, scale) shouldEqual Decimal(rescaled)
+        }
+      }
+    }
+    val dec64WithNull = Array(null) ++ dec64Data.map(_.toJavaBigDecimal) ++ Array(null, null)
+    withResource(
+      GpuColumnVector.from(ColumnVector.fromDecimals(dec64WithNull: _*),
+        DecimalType(DType.DECIMAL64_MAX_PRECISION, 9))) { cv: GpuColumnVector =>
+      cv.getRowCount shouldEqual dec64WithNull.length
+      cv.hasNull shouldBe true
+      cv.numNulls() shouldEqual 3
+      val (precision, scale) = cv.dataType() match {
+        case dt: DecimalType => (dt.precision, dt.scale)
+      }
+      withResource(cv.copyToHost()) { hostCv: RapidsHostColumnVector =>
+        dec64WithNull.zipWithIndex.foreach {
+          case (dec, i) if dec == null =>
+            hostCv.getBase.isNull(i) shouldBe true
+          case (dec, i) =>
+            val rescaled = dec.setScale(scale)
+            hostCv.getLong(i) shouldEqual rescaled.unscaledValue().longValueExact()
+            hostCv.getDecimal(i, precision, scale) shouldEqual Decimal(rescaled)
+        }
+      }
+    }
+    // assertion error throws while running `typeConversionAllowed` check
+    assertThrows[AssertionError] {
+      withResource(GpuColumnVector.from(ColumnVector.decimalFromLongs(0, 1L),
+        DecimalType(DType.DECIMAL64_MAX_PRECISION + 1, 0))) { _ => }
+    }
+    assertThrows[AssertionError] {
+      withResource(GpuColumnVector.from(ColumnVector.decimalFromInts(0, 1),
+        DecimalType(DType.DECIMAL32_MAX_PRECISION + 1, 0))) { _ => }
+    }
+    // TODO: support fromScalar(cudf.ColumnVector cv, int rows) for fixed-point decimal in cuDF
```

Comment:
This is going to be a big problem.  We need this for a project and other operations to work.

Reply:
I will support this feature ASAP. And I really want to know what other missing essentials for this PR. Thanks!

Reply:
What we really have to have/test is 

1) Shuffle decimal types including Contiguous Split, UCX Headers Updated, JCudfSerialzation.  I think all of these have been updated, but not sure if they have been tested yet.
2) CoalesceBatch for decimal types.  This includes Contiguous Split and Table.concat.  Again I think these are in place, but it needs to be tested.
3) ColumnarToRow and RowToColumnar.  I think that is covered here for the most part, but it is not tested.  I would like to have a follow on issue to try and "fix" some of the issues with the fast fixed width transfers, which I am happy to handle.
4) `fromScalar` support. This is something that we are missing but I didn't realize how important it is until recently.  We have a lot of Expressions that require this so we can take a constant value and turn it into a column, which happens regularly.

Reply:
Hi @revans2 , thanks for outlining all essential work for basic decimal supporting. I am trying to address them these days, but I found it is not easy to accomplish all these targets at one time. 
Here is a summary of current progress:
* Enable decimal type for valid `RapidsMeta` in `GpuOverrides` (according to latest supportedType mechanism).
* support decimal for `GpuColumnVector` (with test cases in DecimalUnitTest.scala)
* support decimal for `GpuScalar`  (with test cases in DecimalUnitTest.scala)
* decimal converters for `GpuRowToColumnarExec` (with test cases in GpuBatchUtilsSuite.scala)
* decimal columnar copy for `HostColumnarToGpu` (with test cases in DecimalUnitTest.scala)
* test cases for decimal in `GpuCoalesceBatchesSuite` (Some cases fails because contiguous split of decimal column has yet supported by cuDF. I've created a [PR](https://github.com/rapidsai/cudf/pull/6762) to solve this problem.)
* Creating decimal column vector from scalar will be supported after this [PR](https://github.com/rapidsai/cudf/pull/6723) merged (which is close to).

Could we get this PR merged at first, in order not to block other work relying on this (such as: binary operation support) ?  Then, we address unsolved issues in other PRs. Perhaps we need to create an issue to summarize the progress of all these items?

Reply:
I am fine with breaking things up, but I really want to see something that is self-contained and tested.

If we drop support for decimal in literal values then we don't need `fromScalar` to work yet.  But any code that we do put in I want to see at least minimal automated tests that it is working.  I trust that you have run tests manually, but there is a lot that is changing right now with nested types etc, and I want to be sure that we have tests to verify that someone else didn't break something.

Reply:
Sorry, I am afraid that I don't get what the "minimal automated tests" should be, besides existed unit tests for decimal. Does it refer to tests with mixed data types, such as `GpuSinglePartitioningSuite`? If it does, I am working on them. I've just put a [PR](https://github.com/rapidsai/cudf/pull/6770) to enable creating decimal column with `Table.TestBuilder` .

Reply:
I've just added tests for _binaryOps_ and _project_.